tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1801,8 +1831,14 @@ static bool CombineBasicTypes(
     switch (uBits)
     {
     case BPROP_BITS8:  *pOutKind = AR_BASIC_UINT8;  break;
-    case BPROP_BITS16: (uEitherFlags & BPROP_MIN_PRECISION) ?
-      *pOutKind = AR_BASIC_MIN16UINT : *pOutKind = AR_BASIC_UINT16; break;
+    case BPROP_BITS16:
+      if (IS_BPROP_16BIT_UINT_NO_MIN_PRECISION(uLeftProps) ||
+        IS_BPROP_16BIT_UINT_NO_MIN_PRECISION(uRightProps)) {
```

Comment:
Why not just `BPROP_MIN_PRECISION & uBothFlags` (with reversed then/else)?  We already are under a path that guarantees both sides are unsigned integers.